pages/_app.tsx

```diff
@@ -19,17 +21,25 @@ const inter = Inter({
 
 // Entry point for the app
 export default function App({ Component, pageProps }: AppProps) {
+  const router = useRouter();
+  const shouldRenderComponents = router.pathname === "/";
```

Comment:
This one breaks the site. Legit URLs like `/language/html` return 404.

Reply:
I've identified the error, and I've also added the URLs for the tags:
```
const shouldRenderComponents = router.pathname === "/" || router.pathname.startsWith('/language/') || router.pathname.startsWith('/tag/');
```